### ISSUE TITLE
Question [File Locating]

### ISSUE THREAD

**Issue by Alpha321** [CloudSixteen/Clockwork#277]
In clockwork base files, they have it where if you put a rank in someones name (RCT, DvL, EpU, SeC, CmD) and I was wonder where I can find this, so I can change the models it changes to automatically, I specifically need the MPF model it sets to for like 05.

**Comment by Alpha321** [comment]
Bump?

**Comment by mynameisshi** [comment]
The file in question is sv_hooks, but don't modify it unless you know explicitly what you're doing. Modifying a core schema file can lead to problems and possibly break your framework. I recommend creating a plugin for it. You can find out how to do that [here](http://forums.cloudsixteen.com/threads/how-to-create-a-plugin.2114/) and [here.](http://forums.cloudsixteen.com/threads/how-to-make-a-plugin-getting-started.14574/)

**Comment by MrMe0w** [comment]
As an addition to the comment above, it's highly recommended for any server owner to learn even some basics of Lua to the level where they don't make any syntax mistakes (e.g. forgot a bracket, forgot to close 'if' statement or a function block, used an uninitialized variable etc) and know how to use gmod wiki (wiki.garrysmod.com).
Any Garry's Mod gamemode requires this knowledge to perform any customization, so learning Lua will be a worthwhile investiment of time in case you're not only planning to create one server. It can also serve as a first step in learning programming in general (if you're interested in that kind of stuff).
It may take quite a while to get yourself comfortable with Lua's syntax and functions, but once you do - any customization that you could possibly think of would be a piece of cake. I'd recommend looking at existing code and trying to modify it to do something else or to add functionality. Sandbox gamemode is a great way to start as it's very lightweight and you have a lot of space to experiment.

tl;dr: Learn Lua.


(sorry for a wall of somewhat unrelated text, I just felt it belonged here)

**Comment by xRJx** [comment]
The ranks are actually in the HL2RP files, not the Clockwork files. Specifically lines 28-43 in sv_hooks.lua. Modifying the schema directly is not recommended nor endorsed. Here's an easy way to do it using a plugin:

- Start creating a plugin using this website (it's a web app made by myself): https://clockworkconstructors.com/construct-plugin.php

- Check the /sh_plugin.lua and /sv_hooks.lua checkbox.

- Use this code in the sv_hooks.lua file that was just created in your plugin:

`-- Called when a player's default model is needed.
function PLUGIN:GetPlayerDefaultModel(player)
	local faction = player:GetFaction();
	
	if (Schema:IsCombineFaction(faction)) then
		if (Schema:IsPlayerCombineRank(player, "ENTER RANK HERE")) then
			return "ENTER MODEL PATH AND MODEL NAME HERE";
		-- Uncomment these lines to add multiple ranks/models:
		--[[elseif (Schema:IsPlayerCombineRank(player, "ENTER RANK HERE")) then
			return "ENTER MODEL PATH AND MODEL NAME HERE";
		elseif (Schema:IsPlayerCombineRank(player, "ENTER RANK HERE")) then
			return "ENTER MODEL PATH AND MODEL NAME HERE";
		elseif (Schema:IsPlayerCombineRank(player, "ENTER RANK HERE")) then
			return "ENTER MODEL PATH AND MODEL NAME HERE";--]]
		end;
	end;
end;`
